sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
Thanks @smichr , I understood the problem. But I do not think there is a need of `simplify` flag , because `checksol` is used to check whether `sol` is a solution of equation `f` == `0`. So we should do `simplify` to cancel terms. No need of asking for `simplify`, isn't it ? If we do simplify in this `attempt` 1 then we don't need of attempt 3, 4, 5, 6, 7 ( means no need of `powsimp`, `cancel`, `expand`, `together`).